tests/test_acasclient.py

```diff
@@ -2296,6 +2296,144 @@ def test046_cmpdreg_admin_crud(self):
+            {
+                "dbProperty": "Lot Salt Abbrev",
+                "defaultVal": None,
+                "required": False,
+                "sdfProperty": "Lot Salt Name"
+            },
+            {
+                "dbProperty": "Lot Salt Equivalents",
+                "defaultVal": None,
+                "required": False,
+                "sdfProperty": "Lot Salt Equivalents"
+            }
+        ]
+
+        SALT_ABBREV = 'HCl'
+        SALT_MOL = "\n  Ketcher 05182214202D 1   1.00000     0.00000     0\n\n  1  0  0     1  0            999 V2000\n    6.9500   -4.3250    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0\nM  END\n"
+        # Get Salt Abbrevs. Treat salts separately since they are not a standard codetable
+        salts = self.client.get_salts()
+        # Create Salt Abbrev
+        if SALT_ABBREV.lower() not in [s['abbrev'].lower() for s in salts]:
+            salt = self.client.create_salt(abbrev=SALT_ABBREV, name=SALT_ABBREV, mol_structure=SALT_MOL)
+            self.assertIsNotNone(salt.get('id'))
+
+        response = self.client.register_sdf(test_047_load_sdf_with_salts_file, "bob",
+                                            mappings)
+        print(response)
+        self.assertIn('report_files', response)
+        self.assertIn('summary', response)
+        self.assertIn('Number of entries processed', response['summary'])
+        return response
+
+    def test_048_warn_existing_compound_new_id(self):
+        """
+        Test for Warning When Uploading A "New" Compound That Has Existing Parent and Gets New ID
+        """
+        test_048_warn_existing_compound_new_id_file_one = Path(__file__).resolve().parent.\
+            joinpath('test_acasclient', 'test_048_warn_existing_compound_new_id.sdf')
+        test_048_warn_existing_compound_new_id_file_two = Path(__file__).resolve().parent.\
+            joinpath('test_acasclient', 'test_048_warn_existing_compound_new_id_two.sdf')
+        mappings = [
+            {
+                "dbProperty": "Parent Corp Name",
+                "defaultVal": None,
+                "required": False,
+                "sdfProperty": "Corporate ID"
+            },
+            {
+                "dbProperty": "Lot Chemist",
+                "defaultVal": "bob",
+                "required": True,
+                "sdfProperty": "Lot Scientist"
+            },
+            {
+                "dbProperty": "Project",
+                "defaultVal": "PROJ-00000001",
+                "required": True,
+                "sdfProperty": "Project Code Name"
+            },
+            {
+                "dbProperty": "Parent Stereo Category",
+                "defaultVal": "mixture",
+                "required": True,
+                "sdfProperty": None
+            },
+        ]
+        response = self.client.register_sdf(test_048_warn_existing_compound_new_id_file_one, "bob",
+                                            mappings)
+        print(response)
```

Comment:
I think you can remove the `print` and you should add some more detailed assertions to check that no error is thrown here.

Reply:
Implemented